### PR TITLE
[dmd-cxx] Fix Issue 19778 - ICE when accessing empty array at compile time

### PR DESCRIPTION
**Pull request by ibuclaw**

### CODE REVIEW

**Comment by jacob-carlborg** [comment]
No header update?

File: test/compilable/test19778.d [hunk 1]
```diff
@@ -0,0 +1,6 @@
+struct S
+{
+    int[] data;
+}
+immutable X = S([]);
+enum len = X.data.length;
```
Comment (JinShil) [inline test/compilable/test19778.d]: Interestingly, this produces a compile-time error in current DMD:  https://run.dlang.io/is/upkuqa

Should this be ported to the DDMD frontend?

**Review by PetarKirov — APPROVED** [review]

**Comment by ZombineDev** [comment]
@JinShil see https://github.com/dlang/dmd/pull/9547.

**Comment by ZombineDev** [comment]
@jacob-carlborg there's no headers to be updated.

**Comment by ibuclaw** [comment]
> @jacob-carlborg there's no headers to be updated.

Indeed, it is a static function afterall.

**Comment by ibuclaw** [comment]
@JinShil - All stuff going in dmd-cxx is a backport from D.

**Pull request merged**